pyccel/ast/cuda.py

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+CUDA Module
+This module provides a collection of classes and utilities for CUDA programming.
+"""
+from pyccel.ast.core import FunctionCall
+
+__all__ = (
+    'KernelCall',
+)
+
+class KernelCall(FunctionCall):
+    """
+    Represents a kernel function call in the code.
+    
+     The class serves as a representation of a kernel
+     function call within the codebase.
+
+    Parameters
+    ----------
+    func : FunctionDef
+        The definition of the function being called.
+
+    args : List[LiteralInteger]
+        List of LiteralInteger passed to the function.
+
+    num_blocks : TypedAstNode
+        The number of blocks. These objects must have a primitive type of `PrimitiveIntegerType`.
+
+    tp_block : TypedAstNode
+        The number of threads per block. These objects must have a primitive type of `PrimitiveIntegerType`.
+
+    current_function : FunctionDef, default: None
+        The function where the call takes place.
+    """
+    __slots__ = ('_num_blocks','_tp_block','_func', '_args')
```

Comment:
`__slots__` is a class variable. This means that this class has **both** `FunctionCall.__slots__` and `KernelCall.__slots__`. So your class now has 2 storage spots for `self._func` and `self._args`:

```suggestion
    __slots__ = ('_num_blocks','_tp_block')
```

Reply:
[162c5e8](https://github.com/pyccel/pyccel-cuda/pull/42/commits/162c5e81e51779dd936a0566cecbdc03fe5efdc7)